csrc/host_ir/jit.h

```diff
@@ -0,0 +1,33 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2024-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#pragma once
+#include <fusion.h>
+#include <host_ir/container.h>
+#include <multidevice/communicator.h>
+#include <utils.h>
+#include <memory>
+namespace nvfuser {
+
+class HostIrJit {
+ public:
+  at::Tensor allocate(
+      const kir::Allocate* allocate,
+      const std::vector<int64_t>& input_sizes,
+      const std::vector<int64_t>& input_strides);
+
+  HostIrJit(
+      hir::HostIrContainer* container,
+      int num_threads = kHostIrJitCompileThreads);
+
+  ~HostIrJit();
```

Comment:
```suggestion
  ~HostIrJit() = default;
```

Reply:
nvm, this is coming from the hidden `LlvmJitImpl` thing and the unique_ptr usage of HostIrJit in executor.cpp.

Reply:
Let's put a comment here noting that we cannot have the definition of the destructor here.